Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -806,6 +806,17 @@
     return std::nullopt;
 }
 
+std::optional<RetainPtr<PDFAnnotation>> UnifiedPDFPlugin::annotationForRootViewPoint(const WebCore::IntPoint& point) const
+{
+    auto pointInDocumentSpace = convertFromPluginToDocument(convertFromRootViewToPlugin(point));
+    auto nearestPageIndex = nearestPageIndexForDocumentPoint(pointInDocumentSpace);
```

Comment:
I think I've seen this pattern enough times to warrant us eventually making something like this:

```cpp
class UnifiedPDFPlugin:
enum class CoordinateSpace : uint8_t { RootView, Plugin, Contents, Document, Page };
std::optional<PDFDocumentLayout::PageIndex> nearestPageIndexForPoint(const WebCore::IntPoint&, CoordinateSpace);

auto UnifiedPDFPlugin::annotationForRootViewPoint(auto& point) const
{
    auto nearestPageIndex = nearestPageIndexForPoint(point, CoordinateSpace::RootView);
```

Reply:
(Out of scope for this PR, but I wanted to write it down for my own reference!)